state/state.go

```diff
@@ -993,6 +994,9 @@ func (st *State) AddApplication(args AddApplicationArgs) (_ *Application, err er
 	if args.Charm == nil {
 		return nil, errors.Errorf("charm is nil")
 	}
+	if len(args.AttachStorage) > 0 && args.NumUnits != 1 {
+		return nil, errors.Errorf("AttachStorage is non-empty, but NumUnits is %d", args.NumUnits)
```

Comment:
Nobody should ever see this (apiserver and api client, and cmd, will all check this, and the cmd provides a nicer error message). Nevertheless I'll do it, doesn't hurt.